### ISSUE TITLE
it seems can't work with the situation that one property has multi literal values

### ISSUE THREAD

**Issue by AnkenWang** [jbarrasa/neosemantics#27]
I got an owl file where subjects have an annotation property probably with many literal values
![image](https://cloud.githubusercontent.com/assets/6365031/26769163/2f2587ee-49df-11e7-820d-6ba780b182a5.png)
and after I imported to neo4j, I got the property only with the last value,
![image](https://cloud.githubusercontent.com/assets/6365031/26769389/2c696dde-49e1-11e7-8728-a1e05666780e.png)
Is there any solution for this? Thanks~

**Comment by jbarrasa** [comment]
Hi! Yes, this is a known issue. Currently, only one would be kept because I'm kind of reluctant to use arrays generically when this (multivalued properties) is something that typically happens very rarely. 

If property values had language tags you can select a specific language by using the 'languageFilter' parameter as described in the README of the repo.
  
There is this open discussion (https://github.com/jbarrasa/neosemantics/issues/18) with some ideas around it. I'd be interested to hear what would be your preference before going ahead with any implementation.

Thanks!

JB.

**Comment by jbarrasa** [comment]
I was thinking, maybe pass a parameter with the list of potentially multi-valued properties. That way only these would be loaded as arrays. Could this work for you? I mean, do you know upfront which datatype properties can be multivalued?

I'm kind of liking it now because it would be generic enough and still flexible. But it, of course, requires some upfront knowledge of the dataset you're loading.

What would you think?

Cheers,

JB.

**Comment by AnkenWang** [comment]
Yes, @jbarrasa  I agree that identifying potentially multi-valued properties by passing a parameter would be generic and flexible, much like the option 2 you mentioned in discussion (#18), and it would also work for me.

Thanks,
Xin

**Comment by lifan0127** [comment]
Is there any solution or walk-around now? The use of multi-value property for synonyms is quite common in biological ontology.

**Comment by ali1k** [comment]
isn't there a mechanism in RDF parser to check automatically if a property has multiple literal values and then turn it to a collection in Neo4j?

**Comment by jbarrasa** [comment]
You can specify what properties are multivalued when you call `importRDF` and they will be treated as arrays in Neo4j.
Here's how: 
`CALL semantics.importRDF("http:// ... multival.ttl","Turtle", { handleMultival: 'ARRAY', multivalPropList : ['http://abc.com/voc/prop1','http://abc.com/voc/prop2']})`

I'm busy writing the doc on how to do it. Watch this space...
...or have a look at the unit tests for examples in the meantime.

JB.
